agreement_legal/models/agreement.py

```diff
@@ -0,0 +1,412 @@
+         template field.""",
+    )
+
+    # compute the dynamic content for mako expression
+    def _compute_dynamic_description(self):
+        MailTemplates = self.env["mail.template"]
+        for agreement in self:
+            lang = agreement.partner_id.lang or "en_US"
+            description = MailTemplates.with_context(lang=lang)._render_template(
+                agreement.description, "agreement", [agreement.id]
+            )
+            agreement.dynamic_description = description
+
+    def _compute_dynamic_parties(self):
+        MailTemplates = self.env["mail.template"]
+        for agreement in self:
+            lang = agreement.partner_id.lang or "en_US"
+            parties = MailTemplates.with_context(lang=lang)._render_template(
+                agreement.parties, "agreement", [agreement.id]
+            )
+            agreement.dynamic_parties = parties
+
+    def _compute_dynamic_special_terms(self):
+        MailTemplates = self.env["mail.template"]
+        for agreement in self:
+            lang = agreement.partner_id.lang or "en_US"
+            special_terms = MailTemplates.with_context(lang=lang)._render_template(
+                agreement.special_terms, "agreement", [agreement.id]
+            )
+            agreement.dynamic_special_terms = special_terms
+
+    @api.onchange("field_id", "sub_model_object_field_id", "default_value")
+    def onchange_copyvalue(self):
+        self.sub_object_id = False
+        self.copyvalue = False
+        self.sub_object_id = False
+        if self.field_id and not self.field_id.relation:
+            self.copyvalue = "${{object.{} or {}}}".format(
+                self.field_id.name, self.default_value or "''"
+            )
+            self.sub_model_object_field_id = False
+        if self.field_id and self.field_id.relation:
+            self.sub_object_id = self.env["ir.model"].search(
+                [("model", "=", self.field_id.relation)]
+            )[0]
+        if self.sub_model_object_field_id:
+            self.copyvalue = "${{object.{}.{} or {}}}".format(
+                self.field_id.name,
+                self.sub_model_object_field_id.name,
+                self.default_value or "''",
+            )
+
+    # Used for Kanban grouped_by view
+    @api.model
+    def _read_group_stage_ids(self, stages, domain, order):
+        stage_ids = self.env["agreement.stage"].search(
+            [("stage_type", "=", "agreement")]
+        )
+        return stage_ids
+
+    stage_id = fields.Many2one(
+        "agreement.stage",
+        string="Stage",
+        group_expand="_read_group_stage_ids",
+        help="Select the current stage of the agreement.",
+        tracking=True,
+        index=True,
+    )
+
+    # Create New Version Button
+    def create_new_version(self):
+        for rec in self:
+            if not rec.state == "draft":
+                # Make sure status is draft
+                rec.state = "draft"
```

Comment:
```suggestion
            if not rec.state == "draft":
                # Make sure status is draft
                rec.state = "draft"
```
I am not sure why we need to reset state of rec to "draft".
Or why not just put {"state": "draft"} in to default_vals.